### PR TITLE
RANGER-5005: Use sortpom maven plugin to improve readability of pom files

### PR DESCRIPTION
**Pull request by mneethiraj**
## What changes were proposed in this pull request?
Keeping pom.xml file contents sorted will improve readability. This PR uses sortpom maven plugin to enforce that pom.xml file contents are in sorted order.

## How was this patch tested?
Build with unit tests completed successfully.

### CODE REVIEW

File: plugin-nestedstructure/pom.xml [hunk 1]
```diff
@@ -15,92 +15,90 @@
...
-                <version>15.4</version>
-            </dependency>
-            <dependency>
-                <groupId>org.testng</groupId>
-                <artifactId>testng</artifactId>
-                <version>${testng.version}</version>
-                <scope>test</scope>
-            </dependency>
-        </dependencies>
+    <dependencies>
```
Comment (kumaab) [inline plugin-nestedstructure/pom.xml:40]: A lot of dependencies here use hard-coded versions, may be we should keep them in root pom.xml ?

File: plugin-schema-registry/pom.xml [hunk 2]
```diff
@@ -15,27 +15,25 @@
...
         <version>3.0.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
+    <artifactId>ranger-schema-registry-plugin</artifactId>
+    <packaging>jar</packaging>
+    <name>SchemaRegistry Security Plugin</name>
+    <description>SchemaRegistry Security Plugin</description>
 
     <properties>
         <avro.version>1.11.3</avro.version>
```
Comment (kumaab) [inline plugin-schema-registry/pom.xml:33]: Is it better to keep them in root pom.xml to be consistent?

File: ranger-authn/pom.xml [hunk 3]
```diff
@@ -15,56 +15,36 @@
...
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
     <dependencies>
         <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-common</artifactId>
-            <version>${hadoop.version}</version>
-            <exclusions>
```
Comment (kumaab) [inline ranger-authn/pom.xml:40]: It appears that Exclusions for `hadoop-common` were removed but not added back.

File: ranger-metrics/pom.xml [hunk 4]
```diff
@@ -15,48 +15,31 @@
...
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
     <dependencies>
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-common</artifactId>
-            <version>${hadoop.version}</version>
-            <exclusions>
```
Comment (kumaab) [inline ranger-metrics/pom.xml:41]: Exclusions are removed from here as well.

File: ranger-plugin-classloader/pom.xml [hunk 5]
```diff
@@ -51,4 +50,5 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+    <url>http://maven.apache.org</url>
```
Comment (kumaab) [inline ranger-plugin-classloader/pom.xml]: Minor: Line 53 can be removed ?

File: ugsync/pom.xml [hunk 6]
```diff
@@ -37,42 +73,47 @@
...
                 </exclusion>
             </exclusions>
         </dependency>
         <dependency>
-            <groupId>commons-lang</groupId>
-            <artifactId>commons-lang</artifactId>
-            <version>${commons.lang.version}</version>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-csv</artifactId>
+            <version>1.0</version>
```
Comment (kumaab) [inline ugsync/pom.xml]: minor: can be moved to the root pom.xml.

File: plugin-schema-registry/pom.xml (hunk 2 above)
Comment (mneethiraj) [reply plugin-schema-registry/pom.xml:33]: When a dependency is used only within a plugin module, I suggest keeping its version property in plugin module's pom.xml. For rest of Ranger libraries, version properties should be consolidated in root pom.xml.

File: plugin-nestedstructure/pom.xml (hunk 1 above)
Comment (mneethiraj) [reply plugin-nestedstructure/pom.xml:40]: For dependencies referenced only in a plugin module, I suggest keeping their version properties in that plugin module's pom.xml. Converting hardcoded to properties is a good idea.

File: ranger-authn/pom.xml (hunk 3 above)
Comment (mneethiraj) [reply ranger-authn/pom.xml:40]: Exclusions and version defined in root pom.xml (for hadoop-common) will be applied here too. Hence remove them from child modules.

File: ranger-metrics/pom.xml (hunk 4 above)
Comment (mneethiraj) [reply ranger-metrics/pom.xml:41]: Exclusions and version defined in root pom.xml will be applied here. Hence removed them from here.

**Pull request closed without merging**